### PR TITLE
Call caching hashing strategies

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: supportedBackends/sfs/src/test/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategySpec.scala [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import org.specs2.mock.Mockito
+import wdl4s.values.WdlFile
+
+import scala.util.Success
+
+class ConfigHashingStrategySpec extends FlatSpec with Matchers with TableDrivenPropertyChecks with Mockito with BeforeAndAfterAll {
+
+  behavior of "ConfigHashingStrategy"
+
+  val steak = "Steak"
```
Comment (geoffjentry) [inline supportedBackends/sfs/src/test/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategySpec.scala]: were you hungry when you did this? :)

**Comment by geoffjentry** [comment]
👍

File: supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategy.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  override def hash(file: File): Try[String] = {
+    Try(org.apache.commons.codec.digest.DigestUtils.md5Hex(file.path.toAbsolutePath.toString))
+  }
+
+  override val description = "hash file path"
+}
+case class HashFileStrategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+  override protected def hash(file: File): Try[String] = {
+    tryWithResource(() => file.newInputStream) { inputStream =>
+      org.apache.commons.codec.digest.DigestUtils.md5Hex(inputStream)
```
Comment (mcovarr) [inline supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategy.scala]: why fqn?

File: supportedBackends/sfs/src/test/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategySpec.scala [hunk 3]
```diff
@@ -0,0 +1,124 @@
+package cromwell.backend.impl.sfs.config
+
+//import java.io.File
```
Comment (mcovarr) [inline supportedBackends/sfs/src/test/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategySpec.scala]: nix

**Comment by mcovarr** [comment]
:+1:

**Pull request closed without merging**